exec/java-exec/pom.xml

```diff
@@ -701,18 +707,21 @@
       <!--
       Include the os-maven-plugin to get os.detected.classifier
       -->
-      <extension>
-        <groupId>kr.motd.maven</groupId>
-        <artifactId>os-maven-plugin</artifactId>
-        <version>1.5.0.Final</version>
-      </extension>
-    </extensions>
+      <!-- Uncomment the following to get a debug build that allows openssl support -->
```

Comment:
Can it be in a separate profile (openssl) that is disabled by default?

Reply:
How would that work in conjunction with the main pom's profiles? I would want to have both the (apache | mapr | cdh | hdp)  profile and the openssl profile enabled.

Reply:
For example `-P apache,openssl'. Another option will be to enable openssl profile based on a property.

Reply:
How about we forget the whole os dependent stuff and put in all the variants (there are only four) and make the scope for all of them `provided`?
Since netty is internally loading the correct jar based on the os, all we need to do is make sure that the jar files are in the classpath.

Reply:
Looking at netty code it seems to use `Class.forName("io.netty.internal.tcnative.SSL", false, OpenSsl.class.getClassLoader());` ([see OpenSsl](https://github.com/netty/netty/blob/4.1/handler/src/main/java/io/netty/handler/ssl/OpenSsl.java)), so if all netty-tcnative OS dependent jars are on the classpath, it should load the first in the classpath (if delegated to the system classloader) or one that OpenSsl classloader finds. My understanding is that OpenSsl class is loaded by the system classloader (I may be wrong), but in this case, having all variants of netty-tcnative on the classpath, will not resolve the issue as netty will try to load "io.netty.internal.tcnative.SSL" class only from one of the jars and if it happens to be a wrong jar, will disable OpenSsl functionality.

Reply:
OK. So I tried to put the commented out code in a 'openssl' profile and discovered that maven does not allow the <extensions> tag inside a profile. Since the problem is essentially caused by this extension being included, not being able to put it in the 'openssl' profile makes this useless.
There is a sort of a convoluted hack around this described here: https://stackoverflow.com/questions/17639778/maven-3-profile-with-extensions. 
IMO this introduces a bit of unreadable, magical stuff in the pom which I would rather avoid. 
What do you think?

Reply:
Sounds like a hack. I'd recommend to keep it commented out or try `.mvn/extensions.xml` solution.

Reply:
Updated the PR with the latest recommendations. Using a different profile seems to work well.